### ISSUE TITLE
API references static Vapor instance.

### ISSUE THREAD

**Issue by roflmuffin** [scholtzm/vapor#2]
All new instantiations of API reference the first instance of Vapor created. This clashes heavily when multiple bots are created.

For instance if I create two Vapor clients like so
```js

plugin = {
  name: 'testplugin',
  plugin: function(VaporAPI) {
    var log = VaporAPI.getLogger()

    VaporAPI.registerHandler({
      emitter: 'client',
      event: 'logOnResponse'
    },
    function(logOnResponse) {
      log.info(VaporAPI.getClient().steamID) // These will output the same SteamID
      log.info(VaporAPI.getConfig().username) // Even though we are in the handler for two separate bots.
    })
  }
}


config = {/* unique config options */}
anotherConfig = {/* different config options */}

var bot = vapor();
bot.init(config);
bot.use(plugin);
bot.connect();

var anotherBot = vapor();
anotherBot.init(anotherConfig);
anotherBot.use(plugin);
anotherBot.connect();

```

When the logOnResponse is received, it will output twice (for each bot) but will show only the steamID for the first bot that was created (using `var config`).

Sorry if this is intended behaviour but it makes implementing self-aware plugins a bit tricky?

Also please excuse any JS syntax issues as I mainly use CoffeeScript.

**Comment by scholtzm** [comment]
Vapor was initially a stand-alone client and to use multiple instances, you would simply run two separate clients (2 node processes) and there would be no clashes. I reworked Vapor in 0.8.0 but I did not test this properly and I will look into this once I get back home.

For now, you could run 1 node process per bot. This adds some advantages but also introduces some problems.

However, I want Vapor to support the use-case you described and this was the initial design idea so the current behaviour is definitely not correct or expected.

**Comment by scholtzm** [comment]
Should be fixed in 273bafba68f9d7fa3e216667bafbde3170431b4f.

I have done some basic testing and it worked correctly. I'll close this issue once you test it out as well.

**Comment by scholtzm** [comment]
I have added new example `multiple-accounts` in 36938c811aefeb596133404cd6f9a364aeca4306.

Using multiple instances should not be a problem now. Will be released in 0.9.0.
